src/global.zig

```diff
@@ -170,19 +171,22 @@ pub const GlobalState = struct {
 
         // Find our resources directory once for the app so every launch
         // hereafter can use this cached value.
-        self.resources_dir = try internal_os.resourcesDir(self.alloc);
-        errdefer if (self.resources_dir) |dir| self.alloc.free(dir);
+        self.resources_dir = rd: {
+            if (@hasDecl(apprt.runtime, "resourcesDir")) break :rd try apprt.runtime.resourcesDir(self.alloc);
```

Comment:
I really don't like the use of `@hasDecl` here... Perhaps we should just force every apprt to define `resourcesDir`, and for apprts that don't do any sort of special handling, they could just declare `resourcesDir` to be an alias of `internal_os.resourcesDir`:

```zig
pub const resourcesDir = internal_os.resourcesDir;
```

Reply:
Yea, resources should probably be managed by apprts. Pushed the change.